packages/hash/subgraph/src/stdlib/element/property-type.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  BaseUri,
+  extractBaseUri,
+  extractVersion,
+  VersionedUri,
+} from "@blockprotocol/type-system-node";
+import { OntologyTypeEditionId } from "../../types/identifier";
+import { Subgraph } from "../../types/subgraph";
+import { PropertyTypeWithMetadata } from "../../types/element";
+import { isPropertyTypeVertex } from "../../types/vertex";
+
+/**
+ * Returns all `PropertyTypeWithMetadata`s within the vertices of the subgraph
+ *
+ * @param subgraph
+ */
+export const getPropertyTypes = (
+  subgraph: Subgraph,
+): PropertyTypeWithMetadata[] => {
+  return Object.values(
+    Object.values(subgraph.vertices).flatMap((versionObject) =>
+      Object.values(versionObject)
+        .filter(isPropertyTypeVertex)
+        .map((vertex) => vertex.inner),
+    ),
+  );
+};
+
+/**
+ * Gets a `PropertyTypeWithMetadata` by its `VersionedUri` from within the vertices of the subgraph. Returns `undefined`
+ * if the property type couldn't be found.
+ *
+ * @param subgraph
+ * @param propertyTypeId
+ * @throws if the vertex isn't a `PropertyTypeVertex`
+ */
+export const getPropertyTypeById = (
+  subgraph: Subgraph,
+  propertyTypeId: VersionedUri,
+): PropertyTypeWithMetadata | undefined => {
+  const [baseUri, version] = [
+    extractBaseUri(propertyTypeId),
+    extractVersion(propertyTypeId),
+  ];
```

Comment:
Nit: I think it's a bit odd to create an array just to destructure it, perhaps consider:
```suggestion
  const baseUri = extractBaseUri(propertyTypeId);
  const version = extractVersion(propertyTypeId);
```

Reply:
I personally like it here because it's like you're destructuring `propertyTypeId` into two parts